core/src/mender-client.c

```diff
@@ -1170,15 +1170,18 @@ mender_client_download_artifact_callback(char *type, cJSON *meta_data, char *fil
     mender_err_t ret = MENDER_FAIL;
 
 #if CONFIG_MENDER_LOG_LEVEL >= MENDER_LOG_LEVEL_INF
-    static size_t download_progress = 0;
-    /* New update */
-    if (0 == index) {
-        download_progress = 0;
-    }
-    /* Update every 10% */
-    if (((index * 10) / size) > download_progress) {
-        download_progress = (index * 10) / size;
-        mender_log_info("Downloading '%s' %zu0%%... [%zu/%zu]", type, download_progress, index, size);
+    if (0 < size) {
```

Comment:
I'd do `size > 0`. This is not a comparison that could be confused with an assignment, no need to have the constant on the left. But YMMV.